src/main/java/io/github/patternatlas/api/rest/model/shared/CommentModel.java

```diff
@@ -0,0 +1,84 @@
+package io.github.patternatlas.api.rest.model.shared;
+
+import io.github.patternatlas.api.entities.candidate.comment.CandidateComment;
+import io.github.patternatlas.api.entities.candidate.comment.CandidateCommentRating;
+import io.github.patternatlas.api.entities.issue.IssueRating;
+import io.github.patternatlas.api.entities.issue.comment.IssueComment;
+import io.github.patternatlas.api.entities.issue.comment.IssueCommentRating;
+
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+import lombok.NoArgsConstructor;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.UUID;
+
+@NoArgsConstructor
+@Data
+@EqualsAndHashCode(callSuper = false)
+public class CommentModel {
+
+    private UUID id;
+    private UUID userId;
+    private String userName;
+    private String text;
+    private int rating;
+    private Collection<UUID> upVotes = new ArrayList<>();
+    private Collection<UUID> downVotes = new ArrayList<>();
+
+    /**
+     * For Issue Comments
+     */
+    public CommentModel(IssueComment issueComment) {
+        this.id = issueComment.getId();
```

Comment:
In all these classes where ratings are done, can these be generalized?
However, instead of copy paste lines 34 to  38 in lines 59 to 63, you could define them once a dedicated method.


This applies to all such model classes.

Reply:
Will refactor this part and try to avoid instanceof

Reply:
Extracted functionality into seperate methods - a full refactoring of the comment-models and db will be done in #52